WordPress/gutenberg
# Add npm org scope to WordPress dependencies

Related: #941
Related: #1205

This pull request seeks to add the [WordPress npm organization scope](https://www.npmjs.com/org/wordpress) to all [WordPress dependency](https://github.com/WordPress/gutenberg/blob/master/docs/coding-guidelines.md#wordpress-dependencies) imports. It is a further realization of the intent for these top-level folders to be distributed as independent packages.

Other side-benefits include:

- Surfaces pathed imports on WordPress dependencies
- Surfaces circular dependencies
- Surfaces incorrectly specified `wp_register_script` script dependencies
- Eliminates chance of naming conflicts between "WordPress dependencies" and other npm or built-in modules
- Reduces overall minified bundle size by 8.7% (1285kb -> 1124kb, related to first point and #941)

__Implementation notes:__

This will be hellish to rebase, so I'd prefer not to let this one linger.

__Testing instructions:__

This is difficult to test because it touches nearly every file in the application. Verify through `npm run build`, `npm run dev`, `npm test` that no errors occur, and that no regressions or console errors occur by thorough use of the application.
